### ISSUE TITLE
"aci_l3_outside / argument: relation_l3ext_rs_l3_dom_att" is not working correctly.

### ISSUE THREAD

**Issue by twtko** [terraform-providers/terraform-provider-aci#33]
I have tried to use aci_l3_outside resource with relation_l3ext_rs_l3_dom_att argument with Terraform v0.12.24 and provider.aci v0.2.1.

However, this resource couldn't collect parameters of ACI L3out Domain as below.
This is a part of output when I executed plan operation.

```
(snip)
Terraform will perform the following actions:

  # aci_l3_outside.l3out_ospf will be updated in-place
  ~ resource "aci_l3_outside" "l3out_ospf" {
        enforce_rtctrl               = "export"
        id                           = "uni/tn-Tenant10/out-L3out_OSPF"
        name                         = "L3out_OSPF"
        relation_l3ext_rs_ectx       = "VRF1"
      ~ relation_l3ext_rs_l3_dom_att =jsonencode({}) -> "uni/l3dom-L3-out-Domain"
        target_dscp                  = "unspecified"
        tenant_dn                    = "uni/tn-Tenant10"
    }
(snip)
```

**relation_l3ext_rs_l3_dom_att =jsonencode({})**
This doesn't seem to handle the string correctly.

**Comment by nkatarmal-crest** [comment]
@twtko It may be possible that the relation was not created on the APIC side and we had an error? Can you please check if the create operation works as expected?

**Comment by twtko** [comment]
@nkatarmal-crest After running "terraform apply", I confirmed that the L3out Domain parameter was not properly configured.

I confirmed that I can add L3out Domain using aci_rest resource via Terraform as below.

```
resource "aci_rest" "l3out_domain" {
  path = "api/node/mo/uni/tn-${var.aci_tenant_name}${var.aci_pod_id}/out-${var.aci_l3out_name}/rsl3DomAtt.json"
  class_name = "l3extRsL3DomAtt"

  content = {
    "tDn" = "uni/l3dom-L3-out-Domain"
  }
}
```

I attached debug log of Terraform, that has modified only APIC URL and admin password.
- "with_l3_dom_att - new deploy.log" is when newly created.
- "with_l3_dom_att - modify.log" is when after setting L3out domain via APIC GUI.

[with_l3_dom_att - modify.log](https://github.com/terraform-providers/terraform-provider-aci/files/4556286/with_l3_dom_att.-.modify.log)
[with_l3_dom_att - new deploy.log](https://github.com/terraform-providers/terraform-provider-aci/files/4556287/with_l3_dom_att.-.new.deploy.log)

**Comment by nkatarmal-crest** [comment]
@twtko Thanks for raising this issue. I looked at the logs and figured out the issue. Will work on it.

**Comment by devarshishah3** [comment]
@twtko. Can you share all the resources and the config file that you are using for l3_out config? We can make a module out of it for easier consumption

**Comment by nkatarmal-crest** [comment]
@twtko  I have fixed this issue in the v0.2.2 of the provider. Can you please give it a try?

**Comment by twtko** [comment]
Sorry late reply. I confirmed this problem was fixed in v0.2.2 aci provider for Terraform. Thank you a lot for your so quickly work for this case!!
